src/components/Amount.vue

```diff
@@ -1,25 +1,38 @@
 <template>
     <span class="amount" :class="{ approx: showApprox && isApprox }">
         {{ formattedAmount }}
-        <span class="nim">NIM</span>
+        <span class="currency" :class="currency">{{currency}}</span>
     </span>
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator';
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import { FormattableNumber } from '@nimiq/utils';
+type BigInteger = import ('big-integer').BigInteger;
 
 @Component
 export default class Amount extends Vue {
-    // Amount in luna
-    @Prop(Number) public amount!: number;
+    // Amount in smallest unit
+    @Prop({type: [Number, Object]}) public amount!: number | BigInteger;
     // If set takes precedence over minDecimals and maxDecimals
-    @Prop({type: Number, validator: Amount._validateDecimals }) public decimals?: number;
-    @Prop({type: Number, default: 2, validator: Amount._validateDecimals }) public minDecimals!: number;
-    @Prop({type: Number, default: 5, validator: Amount._validateDecimals }) public maxDecimals!: number;
+    @Prop(Number) public decimals?: number;
+    @Prop({type: Number, default: 2}) public minDecimals!: number;
+    @Prop({type: Number, default: 5}) public maxDecimals!: number;
+    @Prop({type: Number, default: 5}) public totalDecimals!: number;
     @Prop({type: Boolean, default: false}) public showApprox!: boolean;
+    @Prop({type: String, default: 'nim'}) public currency!: string;
 
-    private static _validateDecimals(decimals) {
-        return decimals >= 0 && decimals <= 5 && Number.isInteger(decimals);
+    @Watch('minDecimals', {immediate: true})
+    @Watch('maxDecimals', {immediate: true})
+    @Watch('decimals', {immediate: true})
+    private _validateDecimals(decimals) {
+        if (this.decimals !== undefined && decimals !== this.decimals) {
+            // skip validation for minDecimals and maxDecimals if they're overwritten by decimals
+            return;
+        }
+        if (decimals !== undefined && (decimals < 0 || decimals > this.totalDecimals || !Number.isInteger(decimals))) {
+            throw new Error('Amount: decimals is not in range');
+        }
     }
```

Comment:
Does changing the validator function to only be a watcher (instead of a Vue prop validator) actually prevent the component from continuing? Or does it just continue now with throwing an error but still render? Or is that on purpose?

Reply:
It does throw. 

`0.01$` could of course be displayed as `0.0100$`, so there might be an argument here of removing these checks, allowing them to be done as Vue prop validators again. 
If it throwing is an issue, I am happy to remove the check and restore them as Vue prop validators.